Source/Core/DolphinQt2/Config/ControllersWindow.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QDialog>
+
+#include <array>
+#include <unordered_map>
+
+#include "Core/HW/SI/SI.h"
+#include "Core/HW/Wiimote.h"
+
+class QCheckBox;
+class QComboBox;
+class QHBoxLayout;
+class QFormLayout;
+class QGroupBox;
+class QLabel;
+class QVBoxLayout;
+class QPushButton;
+class QRadioButton;
+
+class ControllersWindow final : public QDialog
+{
+  Q_OBJECT
+public:
+  explicit ControllersWindow(QWidget* parent);
+
+private slots:
+  void OnWiimoteModeChanged(bool passthrough);
+  void OnWiimoteTypeChanged(int state);
+  void OnGCTypeChanged(int state);
+  void SaveSettings();
+  void UnimplementedButton();
+  void OnBluetoothPassthroughSyncPressed();
+  void OnBluetoothPassthroughResetPressed();
+
+private:
+  void CreateGamecubeLayout();
+  void CreateWiimoteLayout();
+  void CreateAdvancedLayout();
+  void CreateMainLayout();
+  void ConnectWidgets();
+  void LoadSettings();
+
+  // Icons
+  QString m_configure_icon;
+  QString m_gamecube_icon;
+  QString m_wii_icon;
+
+  // Main
+  QVBoxLayout* m_main_layout;
+
+  // Gamecube
+  QGroupBox* m_gc_box;
+  QLabel* m_gc_label;
+  QFormLayout* m_gc_layout;
+  std::array<QComboBox*, 4> m_gc_controller_boxes;
+  std::array<QPushButton*, 4> m_gc_buttons;
+  std::array<QHBoxLayout*, 4> m_gc_groups;
+
+  // Wii Mote
+  QGroupBox* m_wiimote_box;
+  QLabel* m_wii_label;
+  QFormLayout* m_wiimote_layout;
+  std::array<QComboBox*, 4> m_wiimote_boxes;
+  std::array<QPushButton*, 4> m_wiimote_buttons;
+  std::array<QHBoxLayout*, 4> m_wiimote_groups;
+  QRadioButton* m_wiimote_emu;
+  QRadioButton* m_wiimote_passthrough;
+  QPushButton* m_wiimote_sync;
+  QPushButton* m_wiimote_reset;
+  QCheckBox* m_wiimote_continous_scanning;
+  QCheckBox* m_wiimote_real_balance_board;
+  QCheckBox* m_wiimote_speaker_data;
+
+  // Advanced
+  QGroupBox* m_advanced_box;
+  QHBoxLayout* m_advanced_layout;
+  QCheckBox* m_advanced_bg_input;
+
+  static int ToGCMenuIndex(const SerialInterface::SIDevices);
```

Comment:
These can also be made internally linked like the variables, which would allow removing the SI and Wiimote includes from the header.